core/src/main/java/org/apache/carbondata/core/cache/CacheProvider.java

```diff
@@ -121,15 +141,24 @@ private void createDictionaryCacheForGivenType(CacheType cacheType, String carbo
    * @param cacheType
    */
   private void createLRULevelCacheInstance(CacheType cacheType) {
-    CarbonLRUCache carbonLRUCache = null;
-    // if cache type is dictionary cache, then same lru cache instance has to be shared
-    // between forward and reverse cache
-    if (cacheType.equals(CacheType.REVERSE_DICTIONARY) || cacheType
-        .equals(CacheType.FORWARD_DICTIONARY)) {
-      carbonLRUCache = new CarbonLRUCache(CarbonCommonConstants.CARBON_MAX_LEVEL_CACHE_SIZE,
-          CarbonCommonConstants.CARBON_MAX_LEVEL_CACHE_SIZE_DEFAULT);
-      cacheTypeToLRUCacheMap.put(CacheType.REVERSE_DICTIONARY, carbonLRUCache);
-      cacheTypeToLRUCacheMap.put(CacheType.FORWARD_DICTIONARY, carbonLRUCache);
+    boolean isDriver = Boolean.parseBoolean(CarbonProperties.getInstance()
```

Comment:
Its bound to the Cache interface methods
V get(K key) throws CarbonUtilException;
List getAll(List keys) throws CarbonUtilException;